### DISCUSSION TITLE
How to use intf_tops.csh

### DISCUSSION THREAD

**Discussion by RenzoAnc** [gmtsar/gmtsar#408] [Q&A] [answered]
Hello, everyone. I'd like to ask you about a doubt I have. I'm doing SBAS processing with S1 images following two tutorials I add here. 

[Tutorial1.pdf](https://github.com/gmtsar/gmtsar/files/9333796/Tutorial1.pdf)
[Tutorial2.pdf](https://github.com/gmtsar/gmtsar/files/9333797/Tutorial2.pdf)

In both tutorials, they mention I should use **select_pairs.csh** with the _baseline_table.dat_ ([baseline_table.txt](https://github.com/gmtsar/gmtsar/files/9333844/baseline_table.txt)) file created with **preproc_batch_tops.csh data.in dem.grd 1** [mode 1]  (not with mode 2). If I'm wrong, I'd like to know it. As results, the intf.in ([intf.txt](https://github.com/gmtsar/gmtsar/files/9333846/intf.txt)), text ([text.txt](https://github.com/gmtsar/gmtsar/files/9333847/text.txt)) and text2 ([text2.txt](https://github.com/gmtsar/gmtsar/files/9333849/text2.txt)) files are created.

But the next step (**intf_tops.csh one.in batch_tops.config**) mentioned by both is different because the first tutorial indicates I should use a _one.in_ file with the following line: S1_20210515_ALL_F2:S1_20210608_ALL_F2, where the first part is my master image (2021-05-15) and the second part is one of the slave images; while the second tutorial indicates I should use a _one.in_ obtained of _head -1 intf.in > one.in_ command. This last command creates a _one.in_ file with the following line: S1_20210115_ALL_F2:S1_20210208_ALL_F2. So the _one.in_ files mentioned contain different lines. Which _one.in_ file should I use for this purpose (intf_tops.csh)?.

Thanks in advance.

**Comment by Xiaohua-Eric-Xu** [comment, marked as answer]
They should get you the same thing i.e. process one interferogram while create the needed topo_ra.grd. It does not matter which one to process here. 

You could do this by linking the master image PRM and LED files to your topo directory and run `dem2topo_ra.csh` there by hand.

**Reply by RenzoAnc** [reply]
Hello, professor @Xiaohua-Eric-Xu. Thank you for your reply. I understand it. I've taken into account your advice and I continued processing with SBAS. I show an interferogram (result) in the images below.

Phase.png:
![phase-1](https://user-images.githubusercontent.com/105986961/184727857-f864eb6c-6bf1-4e61-8a68-541cc2fd07ce.png)


Phasefilt.png:

![phasefilt](https://user-images.githubusercontent.com/105986961/184727787-b8be3793-9dfd-41bb-a2a1-82549258ccbb.png)


In general, the phase.png results look good, but there're discontinuities in the phasefilt.png (for all interferograms). Possibly, It'll get better if I apply preproc_batch_tops_esd.csh (Interpolated ESD, mode 2) and not only preproc_batch_tops.csh.

Despite this problem, I continued unwrapping the interferograms formed with **intf_tops.csh intf.in batch_tops.config**, but I think there was also a problem with one interferogram. In this interferogram, the result was:



![Caso2](https://user-images.githubusercontent.com/105986961/184724611-57b3f46f-b3dc-4612-8cb5-45a4cb106c68.png)


In the image, you can see this warning: _Warning: Unexpected sustained increase in total cost. Breaking loop_. 
The unwrapping files was formed, but I don't know if this warning affected the results.
The other interferograms didn't have this problem (in the unwrapping step). 

What could be the problem?

Regards.

**Reply by Xiaohua-Eric-Xu** [reply]
The discontinuities are likely from `phasefilt` (Goldstein Filter). When you have decorrelation with a larger filter_wavelength, this likely show up I think. You could also tune the patch size for the Goldsetin filter in `filter.csh` or the exponential parameter alpha. ESD only helps with misregistration along azimuth in which case you have horizontal discontinuities at burst boundaries. 

For snaphu, you'll need to look at the result and check on the solution. It usually comes out fine if the last line come up as `Program snaphu done`.

**Reply by RenzoAnc** [reply]
Thank you for your reply, professor @Xiaohua-Eric-Xu. I'll prove changing the parameters (exponential parameter alpha [-alpha] and patch size [-psize]) in filter.csh to fix it and I'll report you the results. So for Snaphu, I think it's a good result because  _Program snaphu done_ appears (not in the last line, but it appears).
Regards.

**Reply by RenzoAnc** [reply]
Hello, @Xiaohua-Eric-Xu. I've proved changing those two parameters and I got the following (in another interferogram):

1) The changes of alpha parameter [0.5 -> 0.9] didn't remove any discontinuity.

2) The changes of path size improve the result (32 -> 64), but there's still discontinuities in some parts of image. The change of path size (32 -> 128) affects so much the result, so I think it isn't adecuated. I've also proved the change of path size 32 -> 16. I add all the results below.

**Path size 16**
![phasefilt16](https://user-images.githubusercontent.com/105986961/184934546-2e77611e-4630-423a-9289-ce1dfc7218bb.png)


**Path size 32**
![phasefilt_32](https://user-images.githubusercontent.com/105986961/184934556-9cc79491-69dd-426f-a0cc-a6616498cc59.png)

**Path size 64**
![phasefilt_64-1](https://user-images.githubusercontent.com/105986961/184934569-4281cb9f-7043-4cad-a533-e7af24575eb9.png)

I think I should continue the SBAS processing with a path size of 64, but I'm not sure about it. However, I see some parts of rigth side have dissapeared with -psize 64. Will it affect the result?.

I'd like to have a suggestion, professor.

Regards.

**Reply by Xiaohua-Eric-Xu** [reply]
I admit that these are pretty arbitrary numbers to play with. The gaussian filter wavelength also has some influence on these patch-like features. And decimation too. I don't know a magic set of parameters that'll always work. But for a small interferogram like yours, I would use less decimation (or rng_dec and azimuth_dec).

**Reply by RenzoAnc** [reply]
Thank you, professor @Xiaohua-Eric-Xu . I'll prove with rng_dec = 4 and azimuth_dec = 1. I'd like to ask you a last question. Do these problems (discontinuities) affect seriously the results of SBAS technique?. To take into account its relevance for it and later processings with SBAS and DInSAR techniques.

Regards.

**Reply by Xiaohua-Eric-Xu** [reply]
I would say not much. These parts usually happens with decorrelation. The sbas program in GMTSAR uses coherence as weights to make a more robust solution on the time-series.

**Reply by RenzoAnc** [reply]
Thank you for your answer, professor @Xiaohua-Eric-Xu.
